### PR TITLE
Add unit tests for `util/environment`

### PR DESCRIPTION
**Pull request by jacobroufa**

### CODE REVIEW

**Comment by jacobroufa** [comment]
This still needs to properly mock the `fs` methods being used here. The tests might pass CI, but they ought not interfere with the env in a permanent fashion. More to come.

File: tests/unit/src/util/environment.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+	decryptIvName() {
+		assert.equal(environment.decryptIvName(), 'publish_deploy_iv');
+	},
+
+	decryptKeyName() {
+		assert.equal(environment.decryptKeyName(), 'publish_deploy_key');
+	},
+
+	encryptedKeyFile() {
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: 3 tests:

* returns ENCRYPTED_KEY_FILE
* returns value passed
* returns default encrypted key file

File: tests/unit/src/util/environment.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		let filename = 'keyfile';
+		let fileWithExt = filename + '.enc';
+		let keyFileDefault = 'deploy_key.enc';
+
+		delete process.env.ENCRYPTED_KEY_FILE;
+		delete process.env.KEY_FILE;
+		assert.equal(environment.encryptedKeyFile(), keyFileDefault);
+
+		process.env.KEY_FILE = filename;
+		assert.equal(environment.encryptedKeyFile(), fileWithExt);
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: we don't need this test. The same branch logic for `encryptedKeyFile()` is already tested in the above test.

File: tests/unit/src/util/environment.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+		'Running locally: returns true'() {
+			delete process.env.HAS_GIT_CREDENTIALS;
+			delete process.env.TRAVIS_BRANCH;
+			assert.isTrue(module.hasGitCredentials());
+		}
+	},
+
+	hexoRootOverride() {
+		let hexoRoot = 'hexoRoot';
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: const

File: tests/unit/src/util/environment.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+		assert.isFalse(environment.isCronJob());
+	},
+
+	isRunningOnTravis() {
+		delete process.env.TRAVIS_BRANCH;
+		assert.isFalse(environment.isRunningOnTravis());
+	},
+
+	keyFile() {
+		let fileName = 'key_file';
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: const

File: tests/unit/src/util/environment.ts [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+		process.env.TRAVIS_EVENT_TYPE = 'not cron';
+		assert.isFalse(environment.isCronJob());
+	},
+
+	isRunningOnTravis() {
+		delete process.env.TRAVIS_BRANCH;
+		assert.isFalse(environment.isRunningOnTravis());
+	},
+
+	keyFile() {
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: would you split this into two named test cases

File: tests/unit/src/util/environment.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+		let fileName = 'key_file';
+		process.env.KEY_FILE = fileName;
+		assert.equal(environment.keyFile(), fileName);
+
+		delete process.env.KEY_FILE;
+		assert.equal(environment.keyFile(), 'deploy_key');
+	},
+
+	publishMode() {
+		let commit = 'commit';
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: const

File: tests/unit/src/util/environment.ts [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+	keyFile() {
+		let fileName = 'key_file';
+		process.env.KEY_FILE = fileName;
+		assert.equal(environment.keyFile(), fileName);
+
+		delete process.env.KEY_FILE;
+		assert.equal(environment.keyFile(), 'deploy_key');
+	},
+
+	publishMode() {
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: split this into two named tests

File: tests/unit/src/util/environment.ts [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+		process.env.DEPLOY_DOCS = commit;
+		assert.equal(environment.publishMode(), commit);
+
+		delete process.env.DEPLOY_DOCS;
+		process.env.TRAVIS_BRANCH = 'master';
+		assert.equal(environment.publishMode(), 'skip');
+	},
+
+	repositorySource() {
+		let target = 'target repo';
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: const

File: tests/unit/src/util/environment.ts [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+		process.env.PUBLISH_TARGET_REPO = target;
+		assert.equal(environment.repositorySource(), target);
+
+		delete process.env.PUBLISH_TARGET_REPO;
+		let slug = 'target_repo';
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: const

File: tests/unit/src/util/environment.ts [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+		let commit = 'commit';
+		process.env.DEPLOY_DOCS = commit;
+		assert.equal(environment.publishMode(), commit);
+
+		delete process.env.DEPLOY_DOCS;
+		process.env.TRAVIS_BRANCH = 'master';
+		assert.equal(environment.publishMode(), 'skip');
+	},
+
+	repositorySource() {
```
Comment (devpaul) [inline tests/unit/src/util/environment.ts]: split this into three test cases

**Pull request merged**